src/com/mendix/widget/badge/components/__tests__/Badge.spec.ts

```diff
@@ -0,0 +1,183 @@
+import { shallow } from "enzyme";
+import { DOM, createElement } from "react";
+
+import { Badge, BadgeProps, OnClickProps } from "../Badge";
+
+import { MockContext, mockMendix } from "tests/mocks/Mendix";
+
+describe("Badge", () => {
+    let badgeProps: BadgeProps;
+    const createBadge = (props: BadgeProps) => shallow(createElement(Badge, props));
+    const newBadgeInstance = (props: BadgeProps) => createBadge(props).instance() as Badge;
+    const clickProps: OnClickProps = { onClickType: "doNothing" };
+
+    beforeEach(() => {
+        badgeProps = {
+            badgeOnClick: clickProps,
+            badgeType: "badge",
+            badgeValue: "0",
+            label: "default",
+            style: "default"
+        };
+        window.mx = mockMendix;
+        window.mendix = { lib: { MxContext: MockContext } };
+    });
+
+    it("should render the structure", () => {
+        const badgeComponent = createBadge(badgeProps);
+
+        expect(badgeComponent).toBeElement(
+            DOM.div(
+                {
+                    className: "widget-badge-display widget-badge-link",
+                    onClick: jasmine.any(Function) as any
+                },
+                DOM.span({ className: "widget-badge-text" }, badgeProps.label),
+                DOM.span({ className: "widget-badge badge label-default" }, badgeProps.badgeValue)
+            )
+        );
+    });
+
+    it("with style 'success' should have class 'widget-badge badge label-success'", () => {
```

Comment:
This is to much detail `widget-badge badge` and not part of the test, they are part of the structure test.